Validation/RecoTrack/python/TrackValidation_cff.py

```diff
@@ -195,27 +242,75 @@ def _addSeedToTrackProducers(seedProducers,modDict):
         modName = "seedTracks"+seed
         if modName not in modDict:
             mod = _trajectorySeedTracks.clone(src=seed)
-            globals()[modName] = mod
+            modDict[modName] = mod
         else:
             mod = modDict[modName]
         names.append(modName)
         seq += mod
     return (names, seq)
 
+def _eraPostfix(era):
+    if era == "":
+        return (era, "")
+    return (era, "_"+era)
+_relevantEras = [
+    _eraPostfix(""),
+    _eraPostfix("trackingPhase1"),
+    _eraPostfix("trackingPhase1PU70"),
+]
+_relevantErasAndFastSim = _relevantEras + [_eraPostfix("fastSim")]
+def _translateArgs(args, postfix, modDict):
+    ret = []
+    for arg in args:
+        if isinstance(arg, list):
+            ret.append(_translateArgs(arg, postfix, modDict))
+        else:
+            ret.append(modDict[arg+postfix])
+    return ret
+def _sequenceForEachEra(function, args, names, sequence, modDict, plainArgs=[], modifySequence=None, includeFastSim=False):
+    if sequence[0] != "_":
+        raise Exception("Sequence name is expected to begin with _")
+
+    _eras = _relevantErasAndFastSim if includeFastSim else _relevantEras
+    for eraName, postfix in _eras:
+        _args = _translateArgs(args, postfix, modDict)
+        _args.extend(plainArgs)
+        ret = function(*_args, modDict=modDict)
+        if len(ret) != 2:
+            raise Exception("_sequenceForEachEra is expected to return 2 values, but function returned %d" % len(ret))
+        modDict[names+postfix] = ret[0]
+        modDict[sequence+postfix] = ret[1]
+
+    # The sequence of the first era will be the default one
+    defaultSequenceName = sequence+_relevantEras[0][0]
+    defaultSequence = modDict[defaultSequenceName]
+    modDict[defaultSequenceName[1:]] = defaultSequence # remove leading underscore
+
+    # Optionally modify sequences before applying the era
+    if modifySequence is not None:
+        for eraName, postfix in _relevantEras:
+            modifySequence(modDict[sequence+postfix])
+
+    # Apply eras
+    for eraName, postfix in _relevantEras[1:]:
+        getattr(eras, eraName).toReplaceWith(defaultSequence, modDict[sequence+postfix])
+def _setForEra(module, era, **kwargs):
+    if era == "":
+        for key, value in kwargs.iteritems():
+            setattr(module, key, value)
+    else:
+        getattr(eras, era).toModify(module, **kwargs)
+
 # Validation iterative steps
-(_selectorsByAlgo, tracksValidationSelectorsByAlgo) = _addSelectorsByAlgo(_algos, globals())
-(_selectorsByAlgo_trackingPhase1, _tracksValidationSelectorsByAlgo_trackingPhase1) = _addSelectorsByAlgo(_algosForTrackingPhase1, globals())
-eras.trackingPhase1.toReplaceWith(tracksValidationSelectorsByAlgo, _tracksValidationSelectorsByAlgo_trackingPhase1)
+_sequenceForEachEra(_addSelectorsByAlgo, args=["_algos"], names="_selectorsByAlgo", sequence="_tracksValidationSelectorsByAlgo", modDict=globals())
```

Comment:
@Dr15Jones You might be interested to see how I ended up in reducing copy-paste for different eras.